src/ImageSearchModal.jsx

```diff
@@ -9,244 +9,214 @@ import { Radio } from "@patternfly/react-core/dist/esm/components/Radio";
-
-        Dialogs.close();
-        this.props.downloadImage(selectedImageName, this.state.imageTag, this.state.isSystem);
-    }
-
-    onItemSelected(_, key) {
-        this.setState({ selected: key.split('-').slice(-1)[0] });
-    }
-
-    onSearchTriggered(forceSearch) {
+export const ImageSearchModal = ({ downloadImage, registries, user, userServiceAvailable, systemServiceAvailable }) => {
+    const [searchInProgress, setSearchInProgress] = useState(false);
+    const [searchFinished, setSearchFinished] = useState(false);
+    const [imageIdentifier, setImageIdentifier] = useState('');
+    const [imageList, setImageList] = useState([]);
+    const [imageTag, setImageTag] = useState("");
+    const [isSystem, setIsSystem] = useState(systemServiceAvailable);
+    const [selectedRegistry, setSelectedRegistry] = useState("");
+    const [selected, setSelected] = useState("");
+    const [dialogError, setDialogError] = useState("");
+    const [dialogErrorDetail, setDialogErrorDetail] = useState("");
+
+    let activeConnection = null;
+    const Dialogs = useDialogs();
+
+    const onSearchTriggered = (value, forceSearch) => {
         // When search re-triggers close any existing active connection
-        if (this.activeConnection)
-            this.activeConnection.close();
-        this.setState({ searchFinished: false });
+        activeConnection = rest.connect(client.getAddress(isSystem), isSystem);
+        if (activeConnection)
+            activeConnection.close();
+        setSearchFinished(false);
 
         // Do not call the SearchImage API if the input string  is not at least 2 chars,
         // unless Enter is pressed, which should force start the search.
         // The comparison was done considering the fact that we miss always one letter due to delayed setState
-        if (this.state.imageIdentifier.length < 2 && !forceSearch)
+        if (imageIdentifier.length < 2 && !forceSearch)
             return;
 
-        this.setState({ searchInProgress: true });
+        setSearchInProgress(true);
 
-        this.activeConnection = rest.connect(client.getAddress(this.state.isSystem), this.state.isSystem);
-        let registries = Object.keys(this.props.registries).length !== 0 ? [this.state.registry] : fallbackRegistries;
+        let queryRegistries = fallbackRegistries;
+        const searchValue = value === null ? selectedRegistry : value;
+        if (Object.keys(registries).length !== 0) {
+            if (searchValue === null || searchValue === "") {
+                queryRegistries = registries.search;
+            } else {
+                queryRegistries = [searchValue];
+            }
+        }
         // if a user searches for `docker.io/cockpit` let podman search in the user specified registry.
-        if (this.state.imageIdentifier.includes('/')) {
-            registries = [""];
+        if (imageIdentifier.includes('/')) {
+            queryRegistries = [""];
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1332-20230704-165425-9756b240-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)